contracts/TelfordL1Relayer.sol

```diff
@@ -0,0 +1,135 @@
+        0xD2dB8075693aA6A0D5C026cdf319D517516c1D40; // Address of Telford source on the testnet
+    address public TELFORD_DESTINATION =
+        crossDomainMessenger.xDomainMessageSender(); // Will update it to hardcode
+
+    /* ========== Events ========== */
+
+    event updatedInfo(
+        address bonder,
+        address user,
+        uint256 amount,
+        uint256 transferId
+    );
+    event RetryableTicketCreated(uint256 indexed ticketId);
+
+    /* ========== Modifier / Constructor ========== */
+
+    constructor() public {
+        owner = msg.sender;
+    }
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only owners can perform this operation!");
+        _;
+    }
+
+    modifier onlyTelfordDestination() {
+        require(
+            msg.sender == crossDomainMessengerOptimism &&
+                crossDomainMessenger.xDomainMessageSender() ==
+                TELFORD_DESTINATION,
+            "Only the Telford Destination Contract can perform this operation!"
+        );
+        _;
+    }
+
+    /* ========== Receiving Function ========== */
+
+    /**
+     * @dev Function that is called on by Optimism's messagers contract via Telford's destination contract
+     * @param _bonder The address of the bonder who provided liquidity for the transaction
+     * @param _user The address of the user who is bridging
+     * @param _amount The amount the user is bridging (sub fees for the bonder)
+     * @param _transferId Number to keep track of the transfer in progress
+     */
+
+    function receiveDestinationTransferConfirmation(
+        address _bonder,
+        address _user,
+        uint256 _amount,
+        uint256 _transferId
+    ) external onlyTelfordDestination {
+        bonderAddress = _bonder;
+        userAddress = _user;
+        amount = _amount;
+        transferId = _transferId;
+        emit updatedInfo(_bonder, _user, _amount, _transferId);
+        relayToArbitrum(10000000000000000, 0, 0);
+    }
+
+    /* ========== Sending Function ========== */
+
+    /**
+     * @dev Function that is called on the final step of accepting the distribution from Telford's Optimism contract,
+     * It's purpose is to send a Retryable Ticket which will call on Telford's Arbitrum fundsReceivedOnDestination function
+     * @param maxSubmissionCost Max gas deducted from user's L2 balance to cover base submission fee
+     * @param maxGas Max gas deducted from user's L2 balance to cover L2 execution
+     * @param gasPriceBid price bid for L2 execution
+     */
+
+    function relayToArbitrum(
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) public payable returns (uint256) {
+        bytes memory data = abi.encodeWithSelector(
+            TelfordSource.fundsReceivedOnDestination.selector
+        );
+        uint256 ticketID = inboxArbitrum.createRetryableTicket{
+            value: msg.value
+        }(
+            TELFORD_SOURCE,
+            0,
```

Comment:
I think this is the `callValue`. I'm not clear what does this `callValue` do. Is it related or does it differ from `msg.value`?

Jason - do you know?